linera-base/src/data_types.rs

```diff
@@ -776,18 +778,18 @@ pub struct Blob {
 }
 
 impl Blob {
-    /// Creates a `HashedBlob` without checking that this is the correct `BlobId`!
-    pub fn with_hash_unchecked(self, blob_id: BlobId) -> HashedBlob {
+    /// Creates a `HashedBlob` without checking that this is the correct `BlobId`.
+    pub fn with_blob_id_unchecked(self, blob_id: BlobId) -> HashedBlob {
         HashedBlob {
             id: blob_id,
             blob: self,
         }
     }
 
-    /// Get the `HashedBlob` from the `Blob`.
-    pub fn into_hashed(self) -> HashedBlob {
-        let id = BlobId::new(&self);
-        HashedBlob { blob: self, id }
+    /// Creates a `HashedBlob` by hashing `self`. No hash checks are made!
+    pub fn with_data_blob_id_unchecked(self) -> HashedBlob {
```

Comment:
Wait, sorry, I mixed this up: This hash _is_ correct, because it's computed right here. What's there to check?

```suggestion
    /// Creates a `HashedBlob` by hashing `self`.
    pub fn with_data_blob_id(self) -> HashedBlob {
```